### DISCUSSION TITLE
Should the caller check for finite coefficients?

### DISCUSSION THREAD

**Discussion by axkr** [Hipparchus-Math/hipparchus#309] [General]
In `LaguerreSolver#solveAllComplex()` should the caller check for definite coefficients or should LaguerreSolver do this test?
in the first case I think it should be mentioned in the Javadoc.

The following example seems to run forever?

```java
public static void main(String[] args) {
    LaguerreSolver laguerreSolver = new LaguerreSolver();
    org.hipparchus.complex.Complex[] solveAllComplex =
        laguerreSolver.solveAllComplex(new double[] {Double.NEGATIVE_INFINITY, 0.0}, 0.0);
  }
```

Suspended stack trace:
![image](https://github.com/Hipparchus-Math/hipparchus/assets/3109793/5fa07e71-0fda-4b9b-9150-e928c34feac0)

**Comment by maisonobe** [comment]
I would consider this as a bug.
It seems `solveAllComplex` calls `setup` with a maximum number of evaluations set to `Integer.MAX_VALUE`, hence it does not detect the infinite loop. We should either use a more reasonable default maximum number of iterations (say 100000 or something like that) or let the user provide this number.

**Reply by axkr** [reply]
What about the coefficients values. 
Should +/- Infinity or NaN values be checked by the method?

**Reply by maisonobe** [reply]
No, I don't think so. There are many reasons why the infinite loop could occur, and the loop is called from several places I think. It would be cumbersome to check each and every one of them. All these cases are contingency cases so testing the loop itself seems fair to me.

**Reply by axkr** [reply]
Ok, should I do "Create issue from discussion"?

**Reply by maisonobe** [reply]
Sure!
